packages/rule-sri/src/sri.ts

```diff
@@ -0,0 +1,305 @@
+    }
+
+    private resources: Map<string, string>;
+    private context: RuleContext;
+    private origin: string;
+    private baseline: string = 'sha384';
+
+    /**
+     * Returns the hash of the content for the given `sha` strengh in a format
+     * valid with SRI:
+     * * base64
+     * * `sha384-hash`
+     */
+    private calculateHash(content: string, sha): string {
+        const hash = crypto
+            .createHash(sha)
+            .update(content)
+            .digest('base64');
+
+        return hash;
+    }
+
+    /**
+     * Checks if the element that originated the request/response is a script or a stylesheet.
+     * There could be other downloads from a `link` element that are not stylesheets and should
+     * be ignored.
+     */
+    private isScriptOrLink(evt: FetchEnd): Promise<boolean> {
+        debug('Is <script> or <link>?');
+        const { element } = evt;
+
+        /*
+         * We subscribe to `fetch::end::script|css`, so element should always exist.
+         * "that should never happen" is the fastest way to make it happen so better be safe
+         */
+        /* istanbul ignore if */
+        if (!element) {
+            return Promise.resolve(false);
+        }
+
+        /*
+         * The element is not one that we care about (could be an img, video, etc.)
+         * No need to report anything, but we can stop processing things right away.
+         */
+        const isScript: boolean = element.nodeName === 'SCRIPT' && !!element.getAttribute('src');
+        const isStyle: boolean = element.nodeName === 'LINK' && element.getAttribute('rel') === 'stylesheet';
+
+        return Promise.resolve(isScript || isStyle);
+    }
+
+    /**
+     * Verifies if the response is elegible for integrity validation. I.E.:
+     *
+     * * `same-origin`
+     * * `cross-origin` on a CORS-enabled request
+     *
+     * More info in https://w3c.github.io/webappsec-subresource-integrity/#is-response-eligible
+     */
+    private async isEligibleForIntegrityValidation(evt: FetchEnd): Promise<boolean> {
+        debug('Is eligible for integrity validation?');
+
+        const { element, resource } = evt;
+        const resourceOrigin: string = new URL(resource).origin;
+
+        // CORS validation only applies to scripts, styles are OK
+        if (element.nodeName !== 'SCRIPT') {
+            return true;
+        }
+
+        if (this.origin === resourceOrigin) {
+            return true;
+        }
+
+        // cross-origin scripts need to be loaded with a valid "crossorigin" attribute (ie.: anonymous or use-credentials)
+        const crossorigin = element.getAttribute('crossorigin');
+
+        if (!crossorigin) {
+            await this.context.report(resource, element, `Cross-origin scripts need a "crossorigin" attribute to be elegible for integrity validation`);
+
+            return false;
+        }
+
+        const validCrossorigin = crossorigin === 'anonymous' || crossorigin === 'use-credentials';
```

Comment:
What if I have `crossorigin="anonymous use-credentials"`? Don't you need to parse the value for it, and then check it it contains a certain value?